Cargo.toml

```diff
@@ -49,10 +49,9 @@ optional = true
 
 [dev-dependencies]
 rand = "0.6"
-rustversion = "1.0"
 static_assertions = "1.1"
-# Pinned to a specific version so that the version used for local development
-# and the version used in CI are guaranteed to be the same. Future versions
-# sometimes change the output format slightly, so a version mismatch can cause
-# CI test failures.
-trybuild = "=1.0.80"
+test-util = { path = "./test-util" }
+
+[[test]]
+name = "ui"
+harness = false
```

Comment:
This is problematic because, in CI, we do `cargo test -- --skip ui` in order to skip this test when running under Miri. With a custom test harness, the command line flags aren't parsed (unless we do it manually), and so `--skip ui` does nothing. Is there any way that we could use the normal test harness and retain the same behavior in this PR?

Reply:
we can make the test itself detect `--skip ui` arguments and just return early. Or we use `#[cfg(miri)]` to bail out immediately